src/apprt/gtk/Window.zig

```diff
@@ -355,6 +360,13 @@ pub fn updateConfig(
     self: *Window,
     config: *const configpkg.Config,
 ) !void {
+    // avoid multiple reconfigs when we have many surfaces contained in this
+    // window using the integer value of config as a simple marker to know if
+    // we've "seen" this particular config before
+    const this_config = @intFromPtr(config);
+    if (self.last_config == this_config) return;
+    self.last_config = this_config;
```

Comment:
This whole thing smells fishy. 

What's the cause of this being called multiple times?

Reply:
It's because the core app loops through every surface and sends a config changed message to the surface. That in turn calls the Window update config function. That happens once for each surface. So if a Window has five splits it will get called five times.

I'm on mobile right now but I'm pretty sure that we do not keep a list of windows anywhere (except buried in GTK's object tree) so we can't send the config changed message to the window except through the individual surfaces.

Reply:
https://github.com/ghostty-org/ghostty/blob/38908e01268298ced33b7b6835e49b8c6d437e12/src/App.zig#L146

Reply:
https://docs.gtk.org/gtk4/method.Application.get_windows.html will give you the list of Windows.